docs/cloud-data-warehouses/snowflake.md

```diff
@@ -1 +1,414 @@
+- When we get into dbt later on, you’ll see how we use dbt to modify the raw data that is in Snowflake to transform and join data together, to build analyses / dashboards off of
+- Both the raw data and the transformed data (from dbt) will be stored here in Snowflake. 
+
+Some examples of how you might be using Snowflake:
+– If you are an owner of one of the data sources, you’ll need to know what the data coming into raw snowflake should look like, and whether it has landed here correctly. You might set up the data connectors that move data into Snowflake, and you might log into Snowflake to examine and explore the data to ensure the data looks accurate and complete
+
+- If you are a project owner or a systems administrator, you might log into Snowflake to understand storage and query costs, give or revoke access to users, or answer any data security questions about the project.
+  
+- If you are an analytics engineer, you will be doing most of your work to transform the data in dbt, but you should understand that whatever data you query and all the queries you build will actually be running in Snowflake.
+    - You’ll also want to come into Snowflake to look at query history for query optimization
+    - You can explore and take a look at data, column names and types, etc. to better understand how to build the final datasets
+      
+- If you are a data engineer, you might use Snowflake's query history to investigate errors in the data pipeline that dbt can not explain
+  
+- If you are an analyst, you can use Snowflake to run adhoc queries of the data and get instant results. You can use SQL and/or Python to examine the data and look for patterns.
+
+- If you are building reports or dashboards, you will be pulling the data for those directly from Snowflake.
+  
+### Snowflake architecture
+
+Snowflake's architecture is unique in that it separates storage and compute resources. It consists of three layers:
+
+![Snowflake architecture diagram](../../images/snowflake_architecture_overview.png) 
+[Image source](https://docs.snowflake.com/en/user-guide/intro-key-concepts)
+
+1. **Cloud services**
+    - Infrastructure manager: manages the set up, monitoring, and maintenance of the Snowflake environment
+    - Optimizer: an intelligent query optimizer that decides the most efficient way to execute queries
+    - Transaction Manager: maintains metadata, schema, and query execution plans
+    - Security: robust security features including role-based access control, user authentication, and encryption
+
+2. **Compute resources (for query processing)**
+    - Executes queries using Massive Parallel Processing (MPP)
+            a cluster = a group of computers or servers that work together as a unified system. they share the workload of a specific task. each cluster operates independently to allow parallel processing[for faster, more efficient processing]. workloads are divided across separate clusters
```

Comment:
Formatting looks a little weird here. Is this supposed to be the start of a new sentence?